### DISCUSSION TITLE
Implement Normalized Mutual Information Measure

### DISCUSSION THREAD

**Discussion by z3y50n** [networkx/networkx#4944] [Algorithms]
As described [here](https://arxiv.org/pdf/cond-mat/0505245.pdf#page=4) this measure can be used to compare two different partitions (one could be a known ground truth).

I would also like to use this as part of the tests on the Louvain Implementation (#4929).

I was wondering if I should create a separate PR for the implementation of the function inside `/networkx/networkx/algorithms/community/quality.py` or include it on the above PR?

**Comment by rossbar** [comment]
> I was wondering if I should create a separate PR for the implementation of the function inside /networkx/networkx/algorithms/community/quality.py or include it on the above PR?

I guess this all depends on the relative size/scope of the various PRs: on the one hand, keeping PRs relatively self-contained *usually* makes them easier to review and leads to them going in faster. OTOH splitting into multiple PRs can lead to complicated inter-PR dependencies where one PR can become a blocker for others etc.

That said, if the MI measure is generally useful in context other than the Louvain community detection, I think introducing it in its own PR makes sense. If that ends up getting bogged down for whatever reason you can always cherry-pick the relevant work back into #4929 :)

**Comment by dschult** [comment]
I agree with @rossbar 
I have a slight preference for keeping the PRs separate especially since I suspect this alternate measure will be relatively fast to review and so it shouldn't slow down the other PR waiting for it to be merged.
